In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split,cross_val_score,GridSearchCV
from sklearn.svm import SVC
import numpy as np

data_train = pd.read_csv("train.csv")
data_test = pd.read_csv("test.csv")

In [25]:
X_train = data_train.iloc[:,0:-1]

In [3]:
y_train = data_train.iloc[:,0:1]

In [20]:
X_test = data_test

In [8]:
svc = SVC()

In [27]:
%%time

svc.fit(X_train,y_train)

D:\anaconda3\lib\site-packages\sklearn\utils\validation.py:63: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  return f(*args, **kwargs)


Wall time: 2min


SVC()

In [29]:
ans0 = svc.predict(X_test)

In [30]:
submission_file0 = pd.read_csv('./sample_submission0.csv')

In [31]:
submission_file0.Label = ans0

In [32]:
submission_file0.to_csv('submission0.csv', index=False)

0.95539

In [33]:
from sklearn.decomposition import PCA

In [77]:
pca = PCA(n_components = 16 ,whiten = True)

In [78]:
print(X_train.shape)
print(X_test.shape)

(42000, 784)
(28000, 784)


In [79]:
X_train_pca = pca.fit_transform(X_train)

In [81]:
X_test_pca = pca.fit_transform(X_test)

In [82]:
print(X_train_pca.shape)
print(X_test_pca.shape)

(42000, 16)
(28000, 16)


In [83]:
svc.fit(X_train_pca,y_train)

D:\anaconda3\lib\site-packages\sklearn\utils\validation.py:63: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  return f(*args, **kwargs)


SVC()

In [84]:
svc.score(X_train_pca,y_train)

0.9765952380952381

In [85]:
ans2 = svc.predict(X_test_pca)

In [86]:
submission_file2 = pd.read_csv('./sample_submission.csv')
submission_file2.Label = ans2
submission_file2.to_csv('submission2.csv', index=False)

n_components_rate n_components   test_score  train_score
 0.9          84         0.49275
 0.85         59         0.51217    0.99424  （whiten =True)
             16        0.49514    0.97660
 

In [87]:
import imblearn

In [88]:
from imblearn.over_sampling import SMOTE

In [89]:
smote = SMOTE()

In [90]:
X2,y2 = smote.fit_resample(X_train,y_train)

In [91]:
%%time

svc.fit(X2,y2)

D:\anaconda3\lib\site-packages\sklearn\utils\validation.py:63: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  return f(*args, **kwargs)


Wall time: 2min 9s


SVC()

In [95]:
%%time 

svc.score(X_train,y_train)

Wall time: 4min 36s


0.9897857142857143

In [93]:
ans3 = svc.predict(X_test)

In [94]:
submission_file3 = pd.read_csv('./sample_submission.csv')
submission_file3.Label = ans3
submission_file3.to_csv('submission3.csv', index=False)

In [ ]:
C=[0.2,0.5,1,2,10]
tol = [1e-4,1e-3,0.01,]
svc = SVC()
clf = GridSearchCV(svc,param_grid={'C':C,'tol':tol})
clf

In [ ]:
%%time
clf.fit(X2,y2)

In [ ]:
clf.score(X_test_pca,y_test)

In [ ]:
clf.best_params_

In [ ]:
import joblib

In [ ]:
joblib.dump(clf,'model')

In [ ]:
clf.score(X_train_pca,y_train)

In [ ]:
X_test_pca.shape

In [ ]:
pre = clf.predict(X_test_pca) 

In [ ]:
pre

In [ ]:
submission_file = pd.read_csv('./sample_submission.csv')

In [ ]:
submission_file.Label

In [ ]:
pre.shape

In [ ]:
submission_file.Label = pre.reshape(:,1)